.test-infra/jenkins/job_sonarqube_report.groovy

```diff
@@ -0,0 +1,52 @@
+/* * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import CommonJobProperties as commonJobProperties
+
+job('beam_sonarqube_report') {
+  commonJobProperties.setTopLevelMainJobProperties(
+        delegate, 'master', 120,
+        true)
+
+  /**
+   * https://issues.jenkins-ci.org/browse/JENKINS-42741
+   */
+  wrappers {
+    withSonarQubeEnv {
+      installationName('ASF Sonar Analysis')
+    }
+  }
+
+  commonJobProperties.setAutoJob delegate
+
+  publishers {
+    archiveJunit('**/build/test-results/**/*.xml')
+  }
+
+  steps {
+    gradle {
+      rootBuildScriptDir(commonJobProperties.checkoutDir)
+      tasks("test")
+      tasks("sonarqube")
+      switches("--continue")
+      switches("-PdisableSpotlessCheck=true")
+
+      // disable parallelization to avoid output collisions
+      switches("--no-parallel")
```

Comment:
Is this necessary? Please file a Jira about it. We should rarely if ever need to disable parallelism. It makes things take an insane amount of time.

Reply:
IIUC, this is not necessary. Gradle will disable the build cache automagically if run in parallel. Not sure about the nitrate here, I think we could live with that until we upgrade to gradle 5 which is supposed to solve this issue.

Reply:
I had to enable this since we override coverage report in different projects. This causes Sonar report and coverage report collide: sonar is reading, while coverage tries to override file.

Reply:
Ouch, Thanks for that hint! But which file is it, which gets overwritten? Because on first sight I would believe that should not happen. So that this reveals some long existing misconfiguration? But  might be totally wrong here.

Reply:
This is a race condition, so I don't have an explicit list. Can run job several times though and see what paths I can get.

Reply:
Ok, I see. Do not know, whether u have the time to do so. Neither if it is worth doing so now. But I have the feeling we should look into that.

Thanks for ur time u spend til now anyway!

Reply:
I guess that safest is to keep it here as of now. I'll see if I can find more clues later.